app/controllers/api/physical_storages_controller.rb

```diff
@@ -20,6 +20,26 @@ def create_resource(_type, _id = nil, data = {})
       action_result(false, err.to_s)
     end
 
+    def delete_resource(type, id, _data = nil)
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" if id.blank?
+
+      api_action(type, id) do |klass|
+        begin
+          physical_storage = resource_search(id, type, klass)
+          unless physical_storage.supports?(:delete)
+            error_msg = "#{physical_storage.type} provider type does not support deleting. #{physical_storage.name} could not be deleted."
```

Comment:
I'm not so sure that we want to override the unsupported error message, this would prevent possibly helpful messages for providers which do support physical_storage deletion but not on a specific instance.

We do `raise record.unsupported_reason(:feature) unless record.supports?(:feature)` in a lot of places and we should continue to for consistency with the rest of the API

Reply:
Ok, I thought it would be nice to have a more informative msg (like I said, if for example we mark 10 storages for deletion but only 1 does not support delete we will get an error msg with the info about the storage that was not deleted and not a general error msg), but I can understand why you want to keep the same logic. I will change it back to raise record.unsupported_reason(:feature)

Reply:
@agrare what about this:
```
          unless physical_storage.supports?(:delete)
            error_msg = "#{physical_storage.name} raised the following error: " + physical_storage.unsupported_reason(:delete)
            raise error_msg
          end
```
so we will get error msg similar to that in the gui:
![image](https://user-images.githubusercontent.com/53213107/97315937-e5296b00-1871-11eb-9c32-eb07a0e3f2c8.png)

if you still dont like it let me know and I will revert it to just raise unsupported_reason(:delete)